activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -179,6 +196,18 @@ def #{method_name}=(value)             # def includes_values=(value)
       CODE
     end
 
+    # TODO: This is akin to how `Relation::VALUE_METHODS` are defined,
+    # but this does not neatly fit into one of the existing categories.
+    # Maybe we should make a full-fledged `WithClause`
+    def with_is_recursive
+      @values.fetch(:with_is_recursive, false)
+    end
+
+    def with_is_recursive=(value)
+      assert_mutability!
+      @values[:with_is_recursive] = value
+    end
```

Comment:
I am not very comfortable with this, but I am not sure what the preferred way to implement this flag would be.

Reply:
I think it should be a simple instance variable like `@async`, `@none` etc.